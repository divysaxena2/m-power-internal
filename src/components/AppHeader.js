import React, { Component } from 'react';
import getTheme from '../../native-base-theme/components';
import material from '../../native-base-theme/variables/material';
import { StyleSheet, View, Image } from 'react-native';
import { Button, StyleProvider, Input, Header, Icon, Right, Left, Body, InputGroup } from 'native-base';


type Props = {};
export default class AppHeader extends Component<Props> {
    render() {
        return (
            <StyleProvider style={getTheme(material)}>
                <Header style={{ paddingTop: 43.12, paddingBottom: 14.39 }} >
                    <View>
                        <Button transparent>
                            <Icon name="ios-contact" style={{ color: '#fff', fontSize: 27 }} />
                        </Button>
                    </View>
                    <View style={styles.searchSection}>
                        <Input placeholder="Tell me what you want to do?" placeholderTextColor="#fff" style={styles.searchBar} />
                        <Icon name="ios-search" style={styles.searchIcon} />
                    </View>
                    <View style={{ paddingLeft: 10.49, paddingRight: 10 }}>
                        <Button transparent>
                            <Image source={require('../assets/scanner.png')} />
                        </Button>
                    </View>
                </Header>
            </StyleProvider>)
    }
}
const styles = StyleSheet.create({
    searchSection: {
        width: 272.51,
        height: 38,
        flexDirection: 'row',
        backgroundColor: '#015BA8',
        borderRadius: 5,
        justifyContent: 'center',
        alignItems: 'center'
    },
    searchBar: {
        color: '#fff',
    },
    searchIcon: {
        color: '#fff',
        fontSize: 12.5,
        paddingRight: 14.84,
        paddingLeft: 5.47

    }
});