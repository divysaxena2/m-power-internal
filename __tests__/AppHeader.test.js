import 'react-native';
import React from 'react';
import AppHeader from '../src/components/AppHeader';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <AppHeader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});