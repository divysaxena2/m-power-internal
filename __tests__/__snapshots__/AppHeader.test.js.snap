// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#004987",
          "borderBottomColor": "#3F51B5",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 101,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 1.2,
          "top": 0,
        },
        Object {
          "paddingBottom": 14.39,
          "paddingTop": 43.12,
        },
      ]
    }
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#3F51B5",
            "borderLeftWidth": null,
            "borderRadius": 2,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#015BA8",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 38,
          "justifyContent": "center",
          "width": 272.51,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        placeholder="Tell me what you want to do?"
        placeholderTextColor="#fff"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            Object {
              "color": "#fff",
            },
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
      />
      <Text />
    </View>
    <View
      style={
        Object {
          "paddingLeft": 10.49,
          "paddingRight": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#3F51B5",
            "borderLeftWidth": null,
            "borderRadius": 2,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
    </View>
  </View>
</View>
`;
